internal/currentthread/thread-id.go

```diff
@@ -0,0 +1,30 @@
+// Package currentthread gives you access to the underlying thread id.
+package currentthread
+
+// //
+// // Extracted from TinyCThread, a minimalist, portable, threading library for C
+// //
+//
+// /* Platform specific includes */
+// #if defined(_WIN32) || defined(__WIN32__) || defined(__WINDOWS__)
+//   #include <windows.h>
+//   typedef HANDLE thrd_t;
+// #else
+//   #include <pthread.h>
+//   typedef pthread_t thrd_t;
+// #endif
+//
+// thrd_t thrd_current(void) {
+//   #if defined(_WIN32) || defined(__WIN32__) || defined(__WINDOWS__)
+//     return GetCurrentThread();
+//   #else
+//     return pthread_self();
+//   #endif
+// }
+// size_t getCurrentTheradID() { return (size_t)thrd_current(); }
+import "C"
+
+// ID returns the id of the current thread
+func ID() int64 {
+	return (int64)(C.getCurrentTheradID())
```

Comment:
wouldn't `syscall.Gettid()` workd?

Reply:
I'm not sure `syscall.Gettid()` works on windows, I have tested this under wine, using [tdm-gcc](https://jmeubank.github.io/tdm-gcc/) and it worked.
I'm a but scared of `syscall.Gettid()`, if you have more experience with it, and are sure it is working well under windows, I'm all good to use it.
Ref:
https://github.com/golang/go/issues/18590#issuecomment-271632152
https://github.com/golang/go/issues/18590#issuecomment-362344496